ldaptor/protocols/ldap/distinguishedname.py

```diff
@@ -124,23 +110,27 @@ def __init__(self, stringValue=None, attributeType=None, value=None):
         if stringValue is None:
             assert attributeType is not None
             assert value is not None
-            self.attributeType = attributeType
-            self.value = value
+            self.attributeType = _to_unicode(attributeType)
+            self.value = _to_unicode(value)
         else:
             assert attributeType is None
             assert value is None
 
-            if isinstance(stringValue, bytes):
-                equal_char = b'='
-            else:
-                equal_char = '='
+            stringValue = _to_unicode(stringValue)
 
-            if equal_char not in stringValue:
+            if '=' not in stringValue:
+                if six.PY2:
+                    stringValue = _to_bytes(stringValue)
```

Comment:
why do we use `_to_bytes`
Once ported, the code should always handle internal data as Unicode.

That is in Py2 and Py3 we will always handle internal data as Unicode.